pkg/platform/abstract/platform.go

```diff
@@ -373,34 +373,43 @@ func (ap *Platform) functionBuildRequired(createFunctionOptions *platform.Create
 }
 
 func (ap *Platform) GetProcessorLogsAndBriefError(scanner *bufio.Scanner) (string, string) {
-	var formattedProcessorLogs, briefErrorMessage string
+	var formattedProcessorLogs, briefErrorsLog string
+
+	briefErrorsArray := &[]string{}
+
+	// adds log to briefErrorsArray (prevents duplicate lines)
+	addToBriefErrorsLog := func(logLine string) {
+		for _, briefError := range *briefErrorsArray {
+			if logLine != "\n" && logLine == briefError {
+				return
+			}
+		}
+		*briefErrorsArray = append(*briefErrorsArray, logLine)
+	}
 
 	for scanner.Scan() {
-		currentLogLine, logLevelAboveInfo, err := ap.prettifyProcessorLogLine(scanner.Bytes())
+		currentLogLine, err := ap.prettifyProcessorLogLine(scanner.Bytes(), addToBriefErrorsLog)
 		if err != nil {
+			rawLogLine := scanner.Text()
 
 			// when it is unstructured just add the log as a text
-			formattedProcessorLogs += scanner.Text() + "\n"
-			briefErrorMessage += scanner.Text() + "\n"
+			formattedProcessorLogs += rawLogLine + "\n"
+			addToBriefErrorsLog(rawLogLine)
 			continue
 		}
 
-		if logLevelAboveInfo {
-			briefErrorMessage += currentLogLine + "\n"
-		}
-
 		// when it is a log line generated by the processor
 		formattedProcessorLogs += currentLogLine + "\n"
 	}
 
-	return common.FixEscapeChars(formattedProcessorLogs), common.FixEscapeChars(briefErrorMessage)
+	// create brief errors log as string, and remove double newlines
+	briefErrorsLog = strings.Join(*briefErrorsArray, "\n")
+	briefErrorsLog = strings.Replace(briefErrorsLog, "\n\n\n", "\n\n", -1)
```

Comment:
You can do the duplicate newline removal inside `prettifyProcessorLogLine` and not deal with the extra `\n` you just added (before joining back to 1 string)